### PR TITLE
Add debug intruments for CBSE-16470

### PR DESCRIPTION
**Pull request by jianminzhao**
- Add logs to the critical path to c4query_run
- Add instruments to detect when column values are freed as column is encoded into JSON.

### CODE REVIEW

File: LiteCore/Query/SQLiteQuery.cc [hunk 1]
```diff
@@ -263,6 +263,15 @@ namespace litecore {
         {
             logInfo("Created on {Query#%u} with %llu rows (%zu bytes) in %.3fms",
                 query->objectRef(), rowCount, recording->data().size, elapsedTime*1000);
+
+            if (_iter[0u]) {
```
Comment (pasin) [inline LiteCore/Query/SQLiteQuery.cc:267]: Is this OK if there is nothing in the array?

File: LiteCore/Query/SQLiteQuery.cc [hunk 2]
```diff
@@ -480,10 +499,12 @@ namespace litecore {
                         throw;
                 }
             }
+            LogTo(SQL, "bindParameters ends");
```
Comment (pasin) [inline LiteCore/Query/SQLiteQuery.cc:502]: What is the default log level of the SQL domain? 

@bmeike I don't remember if enabling verbose log from the platform will also enable SQL domain.

File: LiteCore/Support/Logging.cc [hunk 3]
```diff
@@ -57,6 +57,83 @@ struct ScopedSetter {
     }
 };
 
+
+litecore::LogDomain FleeceLogCB::FLC("Fleece");
+std::unordered_map<const void*, fleece::slice> FleeceLogCB::sQueryEnumNew;
+std::unordered_set<const void*> FleeceLogCB::sColumns;
+std::mutex FleeceLogCB::mutex;
+
+void FleeceLogCB::log(const char* msg, int type, const void* pointer) {
```
Comment (pasin) [inline LiteCore/Support/Logging.cc:66]: What is the purpose of this log function? What is your hypothesis?

File: LiteCore/Query/SQLiteQuery.cc (hunk 1 above)
Comment (jianminzhao) [reply LiteCore/Query/SQLiteQuery.cc:267]: It's okay, just there is no columns i

File: LiteCore/Query/SQLiteQuery.cc (hunk 2 above)
Comment (jianminzhao) [reply LiteCore/Query/SQLiteQuery.cc:502]: I did not check. but it definitely work for when I set the callback level to Info

**Comment by borrrden** [comment]
This looks like it will provide some solid information, assuming that the hypothesis of the problem is in the ballpark.

File: LiteCore/Query/SQLiteQuery.cc (hunk 1 above)
Comment (pasin) [reply LiteCore/Query/SQLiteQuery.cc:267]: Oh, this _iter is for the columns. I see.

File: LiteCore/Support/Logging.cc (hunk 3 above)
Comment (pasin) [reply LiteCore/Support/Logging.cc:66]: Now I understand.

**Pull request merged**